python/ray/serve/controller.py

```diff
@@ -381,10 +381,7 @@ def deploy(
             autoscaling_policy = BasicAutoscalingPolicy(autoscaling_config)
         else:
             autoscaling_policy = None
-        if isinstance(deployer_job_id, bytes):
-            deployer_job_id = ray.JobID.from_int(
-                int.from_bytes(deployer_job_id, "little")
-            )
```

Comment:
I think `deployer_job_id` might still be bytes if called from the [Java codepath](https://github.com/ray-project/ray/blob/master/java/serve/src/main/java/io/ray/serve/api/ServeControllerClient.java#L148).

Reply:
Great catch! I put the `if` statement back in, and I added a comment.

Reply:
Hm, doesn't seem like the Java CI test is triggered... I'm not sure why or how to get it to run 😕

Reply:
@sihanwang41 found today that they aren't triggered by Serve code changes (which is a bug). I believe he's filing a PR to fix this.

Reply:
Hi @shrekris-anyscale, do you mind directly adding RAY_CI_JAVA_AFFECTED = 1 in  https://github.com/ray-project/ray/blob/master/ci/pipeline/determine_tests_to_run.py#L181 . I am not 100% percent sure it can fix (only based on ci code reading), you can directly help to verify :)

Reply:
Nice catch! I added it. Let's see if it kicks off the Java test.

Reply:
Looks like it kicked off the [Java test](https://buildkite.com/ray-project/oss-ci-build-pr/builds/10252#0185e63e-7613-49ed-8482-e0001e0299d4).